src/enlyze/client.py

```diff
@@ -139,58 +154,103 @@ def get_timeseries(
-            )
+        start, end, appliance_uuid = validate_timeseries_arguments(
+            start, end, variables
+        )
+
+        pages = self._client.get_paginated(
+            "timeseries",
+            api_models.TimeseriesData,
+            params={
+                "appliance": appliance_uuid,
+                "start_datetime": start.isoformat(),
+                "end_datetime": end.isoformat(),
+                "variables": ",".join(str(v.uuid) for v in variables),
+            },
+        )
+
+        timeseries_data = _get_timeseries_data_from_pages(pages)
+        if timeseries_data is None:
+            return None
+
+        return timeseries_data.to_user_model(
+            start=start,
+            end=end,
+            variables=variables,
+        )
+
+    def get_timeseries_with_resampling(
+        self,
+        start: datetime,
+        end: datetime,
+        variables: Mapping[user_models.Variable, user_models.ResamplingMethod],
+        resampling_interval: int,
+    ) -> Optional[user_models.TimeseriesData]:
+        """Get timeseries data of :ref:`variables <variable>` for a given time frame.
+
+        Timeseries data for multiple variables can be requested at once. However, all
+        variables must belong to the same appliance.
 
-        start = _ensure_datetime_aware(start)
-        end = _ensure_datetime_aware(end)
+        You should always pass :ref:`timezone-aware datetime
+        <python:datetime-naive-aware>` objects to this method! If you don't, naive
+        datetime objects will be assumed to be expressed in the local timezone of the
+        system where the code is run.
 
-        if start > end:
-            raise EnlyzeError("Start must be earlier than end")
+        :param start: Beginning of the time frame for which to fetch timeseries data.
+            Must not be before ``end``.
+        :param end: End of the time frame for which to fetch timeseries data.
+        :param variables: The variables for which to fetch timeseries data along with a
+            :class:`~enlyze.models.ResamplingMethod` for each variable.
+        :param resampling_interval: The interval in seconds to resample timeseries data
+            with. Must be greater than or equal
+            :const:`~enlyze.constants.MINIMUM_RESAMPLING_INTERVAL`.
 
-        appliance_uuids = frozenset(v.appliance.uuid for v in variables)
+        :raises: |token-error|
 
-        if not appliance_uuids:
-            raise EnlyzeError("Need to request at least one variable")
+        :raises: |resampling-error|
 
-        if len(appliance_uuids) != 1:
-            raise EnlyzeError(
-                "Cannot request timeseries data for more than one appliance per request"
-            )
+        :raises: |generic-error|
+
+        :returns: Timeseries data or ``None`` if the API returned no data for the
+            request
+
+        """
+        variables_sequence = list(variables.keys())
```

Comment:
nit: There's no inherent need to convert the keys into a list. This is just to please mypy because you chose to annotate `validate_timeseries_arguments()` with `Sequence[Variable]` although `Iterable[Variable]` would have been enough.

Reply:
Obsoleted.